pkg/binding/placement.go

```diff
@@ -124,6 +125,22 @@ func (c *Controller) getPlacementByName(name string) (runtime.Object, error) {
 	return got, nil
 }
 
+func (c *Controller) evaluateBindingPolicies(ctx context.Context) {
+	c.logger.Info("evaluating BindingPolicies")
+	placements, err := c.listPlacements()
+	if err != nil {
+		utilruntime.HandleError(err)
+		return
+	}
+	for _, placement := range placements {
+		err = c.handlePlacement(ctx, placement)
```

Comment:
This call would enqueue all workload objects, which can and should be easily avoided since a change in a managed cluster's labels can only affect the destinations resolution of a placement.

Perhaps controller::handlePlacement can support skipping requeueing workload objects.

--

Another point: everywhere else in the controller, whenever an object should be synced or evaluated, it is enqueued to the workqueue. I think working directly with handlePlacement leaves path for the above optimization, but this deviation from the standard workflow should be explained explicitly.

Reply:
We are in agreement then. I am updating the PR accordingly.